app/controllers/self_service/offices_controller.rb

```diff
@@ -0,0 +1,28 @@
+module SelfService
+  class OfficesController < ApplicationController
+    before_action :authenticate_user!
+    before_action -> { @firm = current_firm }
+    before_action -> { @office = current_office }, only: [:destroy]
+
+    def destroy
+      @office.destroy
+      flash[:notice] = I18n.t('self_service.office_destroy.deleted', postcode: @office.address_postcode)
+
+      redirect_to :back
+    end
+
+    private
+
+    def principal
+      current_user.principal
+    end
+
+    def current_firm
+      Firm.find_by(id: params[:firm_id], fca_number: principal.fca_number)
```

Comment:
If firm 123 is within the self-service area, they could access their offices then tweak the :firm_id and be able to access another firms offices.

`current_firm` should be scoped to `principal`